### PR TITLE
Updated mimumum PHP version to 7.4 and tried to add 8.0 to CI

### PR DESCRIPTION
**Pull request by GeeH**
Signed-off-by: Gary Hockin <gary@hock.in>

|    Q          |   A
|-------------- | ------
| Documentation | no
| Bugfix        | no
| BC Break      | yes
| New Feature   | no
| RFC           | no
| QA            | yes

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -12,19 +12,19 @@ env:
...
         - DEPS=lowest
-    - php: 7.3
+    - php: 7.4
       env:
         - DEPS=latest
         - CS_CHECK=true
         - BENCHMARKS=true
         - TEST_COVERAGE=true
-    - php: 7.4
+    - php: 8.0
```
Comment (Ocramius) [inline .travis.yml:24]: We should have this in allowed failures

File: .travis.yml [hunk 2]
```diff
@@ -12,19 +12,19 @@ env:
...
       env:
         - DEPS=lowest
-    - php: 7.4
+    - php: 8.0
```
Comment (Ocramius) [inline .travis.yml:27]: We should have this in allowed failures

**Review by Ocramius — CHANGES_REQUESTED** [review]

File: .travis.yml (hunk 1 above)
Comment (michalbundyra) [reply .travis.yml]: this is not working yet on travis, we have to use nightly if we want test on PHP 8.0

and of course we need much more changes in the travis to allow it to happen....

File: .travis.yml (hunk 2 above)
Comment (GeeH) [reply .travis.yml]: We're doing it on stream at https://twitch.tv/spabby - come and tell us how.

**Review by Ocramius — APPROVED** [review]
:ship:

**Comment by michalbundyra** [comment]
@GeeH @Ocramius 
what's the point of adding nightly build if these are not even configured properly?
See travis build:

> - This package requires php ^7.4 but your PHP version (8.0.0-dev) does not satisfy that requirement.

We need to change the process there to install dependencies when using nightly... Otherwise it has no sense at all.

**Pull request merged**